### PR TITLE
fix(addon-editor): reopen dropdown when clicked another link

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Part of #1302

https://user-images.githubusercontent.com/12021443/164423056-f6f7ba98-dd76-49b1-8aca-c36b8ee3bb0e.mov

## What is the new behavior?

https://user-images.githubusercontent.com/12021443/164423335-6bb914a4-b1a7-4383-aa51-25c23e40bc26.mov

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by waterplea — CHANGES_REQUESTED** [review]
There's no need to reopen the dropdown. `EditLink` component gets the link value in constructor:
```ts
        const doc = shadowRootRef || documentRef;
        const selection = doc.getSelection();

        this.url = selection ? this.getHref(selection) : '';
        this.edit = !this.url;
```
You can probably add `@HostListener('selectionchange')` and refresh the url when necessary.

**Comment by splincode** [comment]
@waterplea `You can probably add @HostListener('window:selectionchange') and refresh the url when necessary.`

Sorry I tried but it doesn't work

**Comment by waterplea** [comment]
> @waterplea `You can probably add @HostListener('window:selectionchange') and refresh the url when necessary.`
> 
> Sorry I tried but it doesn't work

I just tested it and it works. `selectionchange` event is dispatched on `document`, not on `window` as I initially thought.

File: projects/addon-editor/components/editor-new/editor-new.component.ts [hunk 1]
```diff
@@ -127,6 +131,7 @@ export class TuiEditorNewComponent
 
     removeLink(): void {
         this.editor?.unsetLink();
+        this.dropdownSelection?.closeDropdownBox();
```
Comment (waterplea) [inline projects/addon-editor/components/editor-new/editor-new.component.ts:134]: Wouldn't that break tuiPure on the toggle? Why is it necessary to manually close the dropdown?

Comment (splincode) [reply projects/addon-editor/components/editor-new/editor-new.component.ts]: reverted

File: projects/addon-editor/components/edit-link/edit-link.component.ts [hunk 2]
```diff
@@ -36,15 +36,13 @@ export class TuiEditLinkComponent {
...
         @Optional()
         @Inject(TUI_DOCUMENT_OR_SHADOW_ROOT)
-        shadowRootRef: DocumentOrShadowRoot | null,
+        private readonly shadowRootRef: DocumentOrShadowRoot | null,
     ) {
-        const doc = shadowRootRef || documentRef;
-        const selection = doc.getSelection();
-
-        this.url = selection ? this.getHref(selection) : '';
+        this.url = this.makeUrl();
```
Comment (waterplea) [inline projects/addon-editor/components/edit-link/edit-link.component.ts:45]: Please move these two to property declaration, don't forget the order

File: projects/addon-editor/components/edit-link/edit-link.component.ts [hunk 3]
```diff
@@ -62,6 +60,17 @@ export class TuiEditLinkComponent {
             : `${this.url.substr(0, START)}...${this.url.substr(this.url.length - END)}`;
     }
 
+    get isViewMode(): boolean {
```
Comment (waterplea) [inline projects/addon-editor/components/edit-link/edit-link.component.ts:63]: Private? Might also be unnecessary completely.

**Review by waterplea — APPROVED** [review]
A few cosmetic comments

File: projects/addon-editor/components/edit-link/edit-link.component.ts (hunk 2 above)
Comment (splincode) [reply projects/addon-editor/components/edit-link/edit-link.component.ts]: fixed

File: projects/addon-editor/components/edit-link/edit-link.component.ts [hunk 4]
```diff
@@ -104,6 +113,12 @@ export class TuiEditLinkComponent {
         this.url = '';
     }
 
+    private makeUrl(): string {
+        const selection = (this.shadowRootRef ?? this.documentRef).getSelection();
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-editor/components/edit-link/edit-link.component.ts]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**